e2e/suites/actions/version-actions.test.ts

```diff
@@ -0,0 +1,251 @@
+/*!
+ * @license
+ * Alfresco Example Content Application
+ *
+ * Copyright (C) 2005 - 2020 Alfresco Software Limited
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { LoginPage, BrowsingPage, FILES, RepoClient, Utils, UploadNewVersionDialog } from '@alfresco/aca-testing-shared';
+import { VersionManagePage } from '../../../projects/aca-testing-shared/src/components/version-manage/version-manager';
+import { Viewer } from '../../../projects/aca-testing-shared/src/components';
+import { browser } from 'protractor';
+
+describe('Version component actions', () => {
+  const versionManagePage = new VersionManagePage();
+  const viewerPage = new Viewer();
+
+  const username = `user-${Utils.random()}`;
+
+  let fileId: string;
+
+  const filesToUpload = [FILES.pdfFile, FILES.docxFile, FILES.xlsxFile, FILES.jpgFile, FILES.docxFile2];
+
+  const apis = {
+    admin: new RepoClient(),
+    user: new RepoClient(username, username)
+  };
+
+  const loginPage = new LoginPage();
+  const page = new BrowsingPage();
+  const { dataTable, toolbar } = page;
+  const uploadNewVersionDialog = new UploadNewVersionDialog();
+  const { searchInput } = page.header;
+
+  beforeAll(async (done) => {
+    await apis.admin.people.createUser({ username });
+    fileId = (await apis.user.upload.uploadFile(filesToUpload[0])).entry.id;
+    await apis.user.shared.shareFilesByIds([fileId]);
+    done();
+  });
+
+  afterAll(async (done) => {
+    await apis.user.nodes.deleteNodeById(fileId);
+    done();
+  });
+
+  describe('on Personal Files', () => {
+    beforeAll(async (done) => {
+      await loginPage.loginWith(username);
```

Comment:
I think you can move the login in the main beforeAll, instead of logging in in each sub-describe